source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -120,6 +131,11 @@ FilterHeadersStatus Filter::encodeHeaders(ResponseHeaderMap& headers, bool end_s
 
   // Depending on processing mode this may or may not be the first message
   openStream();
+  if (processing_complete_) {
```

Comment:
This pattern keeps coming up after `openStream()`. Is it possible to merge it in for safe-by-construction?

Reply:
There are three states -- everything is OK, the stream didn't open and we should continue anyway, and the stream didn't open and we returned an error and should stop processing. But we could return an enum to that effect and that would simplify things -- let me try that.

Reply:
I cleaned up the code a bit by returning an enum with the three states -- I'll resolve this comment and you can re-open if there are further comments.